src/internal_rules.jl

```diff
@@ -447,6 +447,99 @@ function EnzymeRules.reverse(config, func::Const{typeof(\)}, ::Type{RT}, cache,
     return (nothing,nothing)
 end
 
+
+function EnzymeRules.augmented_primal(
+    config,
```

Comment:
Why make this a separate rule instead of just adding support in the existing rule for `\` by expanding the scope of input types to include these?

Reply:
Yeah good question. I think there are various ways to refactor but I would like to keep this for after we made sure we have the correct rule. Also I find the rules with too many conditionals to be quite hard to read and reason about (it is already kind of hard by nature).

Reply:
But isn't the implementation identical to the previous rule?

Reply:
No, [this line](https://github.com/simsurace/Enzyme.jl/blob/41a9f36d9792b4b03054af0d79460e576408ff3e/src/internal_rules.jl#L531) is different than the counterpart in the dense rule and I removed [these lines](https://github.com/simsurace/Enzyme.jl/blob/41a9f36d9792b4b03054af0d79460e576408ff3e/src/internal_rules.jl#L365C1-L381C4).

Reply:
In any case, there are still a lot of problems. The EnzymeTestUtils test seems to test the wrong thing, so we maybe have to overload that for triangular matrices and perhaps will have to for other ones. But I may be mistaken here. The test would pass if the irrelevant entries of the matrix were zero. I don't see why the irrelevant entries of `dA` are non-zero (I added two `@show` to debug). The non-zero ness does not come from my rule. Presumably the shadow is initialized with zeros by EnzymeTestUtils, so where does it come from?

Reply:
I'd still like this to be merged with the other rule to minimize the maintenance burden and bug surface (since more code will be shared).

Reply:
Yeah I understand this. There are some things (I believe) that should be fixed for the other rule so if you don't mind I can make a follow-up PR where I fix this and refactor both rules to hopefully merge them (if it is possible).

Reply:
Tangentially, do you think this rule is the right one to implement even though it is not the innermost call before hitting LAPACK (3-arg `ldiv!` is)? If it isn't, and we change it for the triangular matrices, we would probably not be able to merge these rules anyway. 
If we keep it as is, people calling ldiv! directly don't get any rule.

Reply:
I think this is now obsolete because there is no existing 3-arg `ldiv!` rule implementation.